store/generated_key/v2/eigenda.go

```diff
@@ -155,35 +175,62 @@ func (e Store) Put(ctx context.Context, value []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	return rlp.EncodeToBytes(cert)
+	switch cert.Version() {
+	case coretypes.VersionTwoCert:
+		return nil, fmt.Errorf("EigenDA V2 certs are not supported anymore, use V3 instead")
+
+	case coretypes.VersionThreeCert:
+		eigenDACertV3, ok := cert.(*coretypes.EigenDACertV3)
+		if !ok {
+			return nil, fmt.Errorf("expected EigenDACertV3, got %T", cert)
+		}
+
+		return eigenDACertV3.Serialize(coretypes.CertSerializationRLP)
```

Comment:
Isnt the serialization type enforced by each cert version? So it feels bug-prone that we're doing this here. For example if someone refactors and changes this by mistake.
That's why originally I thought we should have 2 separate serialization functions: `SerializeABI()` and `SerializeBatcherInbox()` (or whatever).

Haven't thought this through in very deep details though.. so maybe we can discuss before making any changes

Reply:
serialization should be determined by a custom encoding byte which we still need to integrate support for! in the interim it makes the most sense to default to RLP